src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,267 @@
+        }    
+        this.setState({errors: errors});
+        return formIsValid;
+    }
+
+    handleChange(field, e){         
+        let fields = this.state.fields;
+        fields[field] = e.target.value;        
+        this.setState({fields});
+    }
+
+    toggleModal(){
+        this.setState({
+            isOpen: !this.state.isOpen
+        });
+    }
+
+    // write to firebase
+    // var newRequest = firebase.database().ref().child("delivery_requests").push();
+    // newRequest.set(deliveryRequest);
+    
+    submitRequest(){
+        
+    }
+
+    createRequest(event) {
+        if(!this.handleValidation()){ 
+            alert('Form has errors');
+        }else{
+            event.preventDefault();
+            var raUid = event.target.receiveingAgency.value;
+            if (!raUid) raUid = null;
+            
+            var dgUid = event.target.delivererGroup.value;
+            if (!dgUid) dgUid = null;
+
+            var durationValue = null;
+            if (event.target.durationType.value === RequestDurationType.DATE) {
+                durationValue = event.target.endDate.value;
+            } else {
+                durationValue = event.target.numRecurrences.value;
+            }
+
+            // create DeliveryRequest object
+            var deliveryRequest = {
+                startDate: event.target.startDate.value,
+                duration:{
+                    type: event.target.durationType.value,
+                    value: durationValue
+                },
+                repeats: event.target.repeats.value,
+                startTime: event.target.startTime.value,
+                endTime: event.target.endTime.value,
+                primaryContact: event.target.primaryContact.value,
+                notes: event.target.notes.value,
+                receiveingAgency: {
+                    uid: raUid,
+                    confirmed: false
+                },
+                delivererGroup: {
+                    uid: dgUid,
+                    confirmed: false
+                },
+                requestTimestamp: Date.now()
+            };
+
+            this.setState({
+                formInfo: deliveryRequest
+            });
+            
+            this.toggleModal();
+        }
+    }
+
+    render() {
+        return (
+            <div className="form" id="recurringForm">
+                <form onSubmit={this.createRequest}>
+                    <div className="info">
+                        <p id="form-heading">Schedule Recurring Pickup</p>
+                        <p className="error">{this.state.errors['endBeforeStart']}</p>
+                        <p className="error">{this.state.errors['startTime']}</p>
+                        <p className="error">{this.state.errors['endTime']}</p>
+                        <p className="error">{this.state.errors['startDate']}</p>
+                        <p className="error">{this.state.errors['time']}</p>
```

Comment:
so if there aren't any values in `this.state.errors`, these will be empty `<p></p>`s?

Reply:
yes